src/Common/Sharding/Integration/Crc32ModShards.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gaming\Common\Sharding\Integration;
+
+use Gaming\Common\Sharding\Exception\ShardingException;
+use Gaming\Common\Sharding\Shards;
+
+final class Crc32ModShards implements Shards
+{
+    /**
+     * @param string[] $shards
+     *
+     * @throws ShardingException
+     */
+    public function __construct(
+        private readonly array $shards
+    ) {
+        if (count($this->shards) === 0) {
+            throw new ShardingException('At least one shard must be specified.');
+        }
+    }
+
+    public function lookup(string $value): string
+    {
+        return $this->shards[crc32($value) % count($this->shards)];
```

Comment:
This algorithm, although not deem necessary for this project, could be replaced by something more smart, e.g. [consistent hashing](https://en.wikipedia.org/wiki/Consistent_hashing). Consistent hashing would reduce the likelihood of games being reassigned to another shard in the event of resharding. A comment on the class which trade-offs this brings should suffice.